jgrapht-core/src/main/java/org/jgrapht/Graphs.java

```diff
@@ -433,6 +435,228 @@
+            if (predicate.test(node)) {
+                verticesToRemove.add(node);
+            }
+        }
+
+        return removeVerticesAndConnectParentsWithChildren(
+            graph,
+            verticesToRemove);
+    }
+
+    /**
+     * Removes all the given vertices from the given graph. If the vertex to be
+     * removed has one or more parents, the parents will be connected directly
+     * to the children of the vertex to be removed.
+     *
+     * @param graph to be mutated.
+     * @param vertices vertices to be removed from this graph, if present.
+     *
+     * @return true if at least one vertex has been removed; false otherwise.
+     */
+    public static <V, E extends DefaultEdge> boolean removeVerticesAndConnectParentsWithChildren(
+        DirectedGraph<V, E> graph,
+        List<V> vertices)
+    {
+
+        boolean atLeastOneVertexHasBeenRemoved = false;
+
+        for (V vertex : vertices) {
+            if (removeVertexAndConnectParentsWithChildren(graph, vertex)) {
+                atLeastOneVertexHasBeenRemoved = true;
+            }
+        }
+
+        return atLeastOneVertexHasBeenRemoved;
+    }
+
+    /**
+     * Add edges from one source vertex to multiple target vertices. Whether
+     * duplicates are created depends on the underlying {@link DirectedGraph}
+     * implementation.
+     *
+     * @param graph graph to be mutated
+     * @param source source vertex of the new edges
+     * @param targets target vertices for the new edges
+     */
+    public static <V, E extends DefaultEdge> void addOutgoingEdges(
+        DirectedGraph<V, E> graph,
+        V source,
+        List<V> targets)
+    {
+        if (!graph.containsVertex(source)) {
+            graph.addVertex(source);
+        }
+        for (V target : targets) {
+            if (!graph.containsVertex(target)) {
+                graph.addVertex(target);
+            }
+            graph.addEdge(source, target);
+        }
+    }
+
+    /**
+     * Add edges from multiple source vertices to one target vertex. Whether
+     * duplicates are created depends on the underlying {@link DirectedGraph}
+     * implementation.
+     *
+     * @param graph graph to be mutated
+     * @param target target vertex for the new edges
+     * @param sources source vertices for the new edges
+     */
+    public static <V, E extends DefaultEdge> void addIncomingEdges(
+        DirectedGraph<V, E> graph,
+        V target,
+        List<V> sources)
+    {
+        if (!graph.containsVertex(target)) {
+            graph.addVertex(target);
+        }
+        for (V source : sources) {
+            if (!graph.containsVertex(source)) {
+                graph.addVertex(source);
+            }
+            graph.addEdge(source, target);
+        }
+    }
+
+    public static <V, E extends DefaultEdge> boolean vertexHasChildren(
+        DirectedGraph<V, E> graph,
+        V vertex)
+    {
+        return !Graphs.successorListOf(graph, vertex).isEmpty();
```

Comment:
Changed. See commit _3aa00a6018dafcb94c46e43a68924769aac6af9b_. As long as there is no such thing as a dangling edge ( `A --->` ) this approach seems fine ;-)